src/libraries/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs

```diff
@@ -149,15 +157,23 @@ public ClientCertificateOption ClientCertificateOptions
                 switch (value)
                 {
                     case ClientCertificateOption.Manual:
+#if TARGETS_BROWSER
+                        _clientCertificateOptions = value;
+#else
                         ThrowForModifiedManagedSslOptionsIfStarted();
                         _clientCertificateOptions = value;
-                        _socketsHttpHandler.SslOptions.LocalCertificateSelectionCallback = (sender, targetHost, localCertificates, remoteCertificate, acceptableIssuers) => CertificateHelper.GetEligibleClientCertificate(ClientCertificates)!;
+                        _underlyingHandler.SslOptions.LocalCertificateSelectionCallback = (sender, targetHost, localCertificates, remoteCertificate, acceptableIssuers) => CertificateHelper.GetEligibleClientCertificate(ClientCertificates)!;
+#endif
                         break;
 
                     case ClientCertificateOption.Automatic:
+#if TARGETS_BROWSER
+                        _clientCertificateOptions = value;
+#else
                         ThrowForModifiedManagedSslOptionsIfStarted();
                         _clientCertificateOptions = value;
-                        _socketsHttpHandler.SslOptions.LocalCertificateSelectionCallback = (sender, targetHost, localCertificates, remoteCertificate, acceptableIssuers) => CertificateHelper.GetEligibleClientCertificate()!;
+                        _underlyingHandler.SslOptions.LocalCertificateSelectionCallback = (sender, targetHost, localCertificates, remoteCertificate, acceptableIssuers) => CertificateHelper.GetEligibleClientCertificate()!;
+#endif
```

Comment:
These ifdefs are ok for now.  It'd be good to see if we can find a cleaner solution subsequently.